vision_agent/agent/vision_agent.py

```diff
@@ -184,6 +193,12 @@ def chat_with_code(
                 or if it contains media files, it should be in the format of:
                 [{"role": "user", "content": "describe your task here...", "media": ["image1.jpg", "image2.jpg"]}]
             artifacts (Optional[Artifacts]): The artifacts to use in the task.
+            test_multi_plan (bool): If True, it will test tools for multiple plans and
+                pick the best one based off of the tool results. If False, it will go
+                with the first plan.
+            customized_tool_names (List[str]): A list of customized tools for agent to
```

Comment:
Is this similar to what @MingruiZhang developed in the UI. Send a custom list of tools to use from the overall set of tools ?

Reply:
Yes this is the same thing. It was added to VisionAgentCoder and I just made sure it worked in VisionAgent.